### ISSUE TITLE
QuietException is shown at INFO logging level

### ISSUE THREAD

**Issue by seliv** [cometd/cometd#1094]
Hello!
In our project we faced an issue with logging `EofException` to the log at `INFO` level which pollutes the log and raises false alarms from the support team. Based on my analysis of the code this must be classified as an incorrect (unwanted) behavior and should be fixed.

Details:

- `org.eclipse.jetty.io.EofException` implements `org.eclipse.jetty.io.QuietException`.
- `org.eclipse.jetty.io.QuietException` is a marker interface whose sole purpose is marking the exceptions that should be logged less verbosely (as the inline javadoc says).
- The logic for logging less verbosely is located at `onError(Throwable failure)` method of `org.cometd.server.websocket.common.AbstractWebSocketEndPoint`:

```
                if (failure instanceof QuietException) {
                    if (_logger.isDebugEnabled()) {
                        _logger.debug("WebSocket failure, address: " + address, failure);
                    }
                } else {
                    _logger.info("WebSocket failure, address: " + address, failure);
                }
```

The intention of this code is clear but the logic fails to execute correctly in a live runtime environment.
The debugger shows that `failure instanceof QuietException` expression evaluates to `false` for `EofException` instance.
It happens because `EofException` (and its parent `QuietException`) on the left of `instanceof` and `QuietException` on the right are loaded with different classloaders. This can be confirmed by looking at `failure.getClass().getClassloader()` and `QuietException.class.getClassloader()`, classloader instances are different.

Please advise us if this is actually a bug that should be fixed within CometD framework or we have some misconfiguration in our environment that causes the issue.

The refined minimal sample to demonstrate the issue can be found here: https://github.com/seliv/cometd-eof-quiet-exception-issue

**Comment by sbordet** [comment]
I cannot reproduce the issue with your sample project. The server logs is always clean (no INFO log lines on the terminal) when I close the browser tab.

**Comment by youngj** [comment]
I encountered the same issue when deploying CometD 7.0.5 to a production environment with WebSocket clients, although I have not reproduced the error locally. I think it happens when WebSocket clients go away due to network connectivity issues rather than just closing the browser tab normally. As a result CometD was logging a large volume of meaningless stack traces, which could not be suppressed by setting `org.eclipse.jetty.LEVEL=WARN` in jetty-logging.properties. 

Eventually I figured out that these errors could be suppressed by creating a simplelogger.properties file in the resources directory, with the following content:
```
org.slf4j.simpleLogger.log.org.cometd.server.websocket.javax.WebSocketEndPoint$Delegate=warn
```
